yellow-paper/docs/public-vm/gen/_InstructionSet.mdx

```diff
@@ -0,0 +1,1240 @@
+	<td><Markdown>Call into another contract, disallowing persistent state modifications.</Markdown></td>
+	<td><Markdown>208</Markdown></td>
+	<td><CodeBlock language="jsx">
+{`M[successOffset] = staticcall(
+    M[l1GasOffset], M[l2GasOffset], M[addrOffset],
+    M[argsOffset], M[argsSize],
+    M[retOffset], M[retSize])`}
+	</CodeBlock></td>
+</tr>
+<tr>
+	<td style={{'text-align': 'center'}}>0x19</td>	<td style={{'text-align': 'center'}}><a id='isa-table-ulog'/><Markdown>[`ULOG`](#isa-section-ulog)</Markdown></td>
+	<td><Markdown>Emit an unencrypted log with data from the `field` memory page</Markdown></td>
+	<td><Markdown>64</Markdown></td>
+	<td><CodeBlock language="jsx">{
+		`ulog(M[offset:offset+size])`
+	}</CodeBlock></td>
+</tr>
+<tr>
+	<td style={{'text-align': 'center'}}>0x1a</td>	<td style={{'text-align': 'center'}}><a id='isa-table-chainid'/><Markdown>[`CHAINID`](#isa-section-chainid)</Markdown></td>
+	<td><Markdown>Get this rollup's L1 chain ID</Markdown></td>
+	<td><Markdown>40</Markdown></td>
+	<td><CodeBlock language="jsx">{
+		`M[dstOffset] = Globals.chainId`
+	}</CodeBlock></td>
+</tr>
+<tr>
+	<td style={{'text-align': 'center'}}>0x1b</td>	<td style={{'text-align': 'center'}}><a id='isa-table-version'/><Markdown>[`VERSION`](#isa-section-version)</Markdown></td>
+	<td><Markdown>Get this rollup's L2 version ID</Markdown></td>
+	<td><Markdown>40</Markdown></td>
+	<td><CodeBlock language="jsx">{
+		`M[dstOffset] = Globals.version`
+	}</CodeBlock></td>
+</tr>
+<tr>
+	<td style={{'text-align': 'center'}}>0x1c</td>	<td style={{'text-align': 'center'}}><a id='isa-table-blocknumber'/><Markdown>[`BLOCKNUMBER`](#isa-section-blocknumber)</Markdown></td>
+	<td><Markdown>Get this block's number</Markdown></td>
+	<td><Markdown>40</Markdown></td>
+	<td><CodeBlock language="jsx">{
+		`M[dstOffset] = Globals.blocknumber`
+	}</CodeBlock></td>
+</tr>
+<tr>
+	<td style={{'text-align': 'center'}}>0x1d</td>	<td style={{'text-align': 'center'}}><a id='isa-table-timestamp'/><Markdown>[`TIMESTAMP`](#isa-section-timestamp)</Markdown></td>
+	<td><Markdown>Get this L2 block's timestamp</Markdown></td>
+	<td><Markdown>40</Markdown></td>
+	<td><CodeBlock language="jsx">{
+		`M[dstOffset] = Globals.timestamp`
+	}</CodeBlock></td>
+</tr>
+<tr>
+	<td style={{'text-align': 'center'}}>0x1e</td>	<td style={{'text-align': 'center'}}><a id='isa-table-coinbase'/><Markdown>[`COINBASE`](#isa-section-coinbase)</Markdown></td>
+	<td><Markdown>Get the block's beneficiary address</Markdown></td>
+	<td><Markdown>40</Markdown></td>
+	<td><CodeBlock language="jsx">{
+		`M[dstOffset] = Globals.coinbase`
+	}</CodeBlock></td>
+</tr>
+<tr>
+	<td style={{'text-align': 'center'}}>0x1f</td>	<td style={{'text-align': 'center'}}><a id='isa-table-blockl1gaslimit'/><Markdown>[`BLOCKL1GASLIMIT`](#isa-section-blockl1gaslimit)</Markdown></td>
```

Comment:
Why do we need this to be available ?

Reply:
Not sure we do, but its an EVM opcode `GASLIMIT`